app/components/Showcases/Barter.jsx

```diff
@@ -0,0 +1,807 @@
+                    placeholder="placeholder"
+                    allowPubKey={true}
+                    allowUppercase={true}
+                    account={to_account}
+                    accountName={to_name}
+                    onChange={this.toChanged.bind(this)}
+                    onAccountChanged={this.onToAccountChanged.bind(this)}
+                    hideImage
+                    typeahead={true}
+                />
+                <AmountSelector
+                    label="showcases.barter.title"
+                    amount={this.state.to_barter[0].to_amount}
+                    onChange={this.onToAmountChanged.bind(this, 0)}
+                    asset={
+                        to_asset_types.length > 0 && to_barter[0].to_asset
+                            ? to_barter[0].to_asset.get("id")
+                            : to_barter[0].to_asset_id
+                                ? to_barter[0].to_asset_id
+                                : to_asset_types[0]
+                    }
+                    assets={to_asset_types}
+                    display_balance={balance(
+                        to_account,
+                        to_balanceError,
+                        to_asset_types
+                    )}
+                    allowNaN={true}
+                />
+                <div style={{paddingTop: "10px", paddingBottom: "10px"}}>
+                    <Button onClick={this.addToAmount.bind(this)}>
+                        + Add asset
+                    </Button>
+                </div>
+                {toAmountSelector}
+            </Card>
+        );
+
+        let offers = (
+            <Card style={{borderRadius: "10px"}}>
+                <div className="left-label">
+                    {from_name} offers {assetFromList.join(", ")} to {to_name}{" "}
+                    and receives {assetToList.join(", ")} in return
+                </div>
+                {from_barter.length === 1 && to_barter.length === 1 ? (
+                    <div className="amount-selector" style={this.props.style}>
+                        <Translate
+                            className="left-label"
+                            component="label"
+                            content="transfer.explict_price"
+                        />
+                        <div className="inline-label input-wrapper">
+                            <input
+                                disabled={false}
+                                type="text"
+                                value={explictPrice()}
+                            />
+
+                            <div className="form-label select floating-dropdown">
+                                <div className="dropdown-wrapper inactive">
+                                    <div>
+                                        {`${assetFromSymbol}/${assetToSymbol}`}
+                                    </div>
+                                </div>
+                                )}
```

Comment:
The code does not compile with this line. Please always compile before committing.

Reply:
> The code does not compile with this line. Please always compile before pushing

Sorry, but it is compiled successfully for me

Reply:
About 
> the current view is broken I think (asset selector missing on the right side)

It is caused because we did not select recipient user. We take assets from user actuall assets list. Maybe we should disable assets field and add Asset button when user is not choosen ?

Reply:
Yes perfect suggestion!

I didn't even notice, apologies

Reply:
I added the disabling method for asset field and addAsset button when an account is not chosen. Also, I fixed the issue with  total amount indicator and user changing ( asset fields did not update during user changing)

Reply:
Also I fixed total Fee calculating method